R/workhorse.R

```diff
@@ -1,78 +1,145 @@
-build_request <- function(endpoint = NULL,
+build_request <- function(path = NULL,
```

Comment:
You might want to consider a two-layer request builder, like I made here:

https://github.com/tidyverse/googlesheets2/blob/8caf705a92b7c0638783ce478eb0449a3a92c1de/R/gs_build_request.R#L47

The inner layer, `gs_build_request()`, is very dumb and mechanical. But would work if someone really knew what they were doing and only wanted help setting base URL, path param expansion, and managing auth.

The outer layer, `gs_generate_request()`, is smarter and what I plan to use! It needs to be given a known endpoint and then it looks up everything we now, checks that input looks good, and massages it into shape before calling `gs_build_request()`.

If you're game, you can make that a separate PR. Let's get this one merged.